k8s-charm/lib/charms/juju_dashboard/v0/juju_dashboard.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, charm: CharmBase, relation: Relation, provider: Application):
         if charm.unit.is_leader():
             # TODO: handle the situation where there are multiple dashes, and the endpoint
             # is the haproxy address.
-            ip = check_output(["unit-get", "public-address"]).decode().strip()
+            ip = str(charm.model.get_binding(relation).network.ingress_address)
```

Comment:
Are you sure about this change? I can see that it was added to the previous PR:
https://github.com/canonical/juju-dashboard-charm/pull/16/commits/fea4a4e13fdb16fe81e7b211fee3e0e8592b9a07

Reply:
Yes I intentionally reverted this change, as it was meant to fix an issue only on LXD (machine) and adding it for k8s is breaking the dashboard and making it inaccessible.